frontend/packages/kubevirt-plugin/src/components/create-vm/forms/boot-source-form.tsx

```diff
@@ -142,14 +142,14 @@ const AdvancedSection: React.FC<AdvancedSectionProps> = ({
             )}
             isChecked={applySP}
             onChange={() => setApplySP(!applySP)}
-            isDisabled={!isSPSettingProvided}
+            isDisabled={!isSPSettingProvided || isSCNameEmpty}
             label={t('kubevirt-plugin~Apply optimized StorageProfile settings')}
             data-test="apply-storage-provider"
           />
         </FormRow>
         {!spLoaded ? (
           <LoadingInline />
-        ) : isSPSettingProvided && applySP ? (
+        ) : isSPDefaults ? (
```

Comment:
Since we're loading the same component for the `!spLoaded` and `!isSPdefaults` conditions, can we remove the nested ternary here and do like below?

```
{!spLoaded || !isSPDefaults ? (
  <LoadingInline />
) : (
    <Everything else>
)}
```
I think it would be even better if we can remove the negations and do this:

```
{spLoaded && isSPDefaults ? (
  <Everything else>
) : (
    <LoadingInline />
)}
```

Reply:
there could be 2 different cases:
1. `spLoaded` is set to false which we are still waiting for data to be fetched from the StorageProfile Resource
2. `spLoaded` is set to true and this also has 2 different cases:
a. `isSPDeafults` is set to `false` which can happen either if the user unchecks the defaults OR there is no default 
values in StorageProfile OR when StorageClass is undefined, in this case, the access mode and volumes mode radio components should be visible to the user.
b. `isSPDeafults` is set to `true` in this case the default values from the StorageProfile should be visible

Reply:
Ah, I thought the false branch of the ternary that starts on line 198 was for the `isSPDefaults` condition. I didn't realize that it doesn't have a false branch. Have I mentioned how much I hate nested ternaries? lol. They're even worse when I can't select a bracket and have the matching bracket highlighted.

Would it work if you use `isSPDefaults && <Everything else>` instead of a ternary with only a true branch? I think that would make it more readable.